File: 0011-container-with-most-water/0011-container-with-most-water.js
/**
 * @param {number[]} height
 * @return {number}
 */
var maxArea = function(height) {
    let l=0, r=height.length-1
    let res = -1
    while(r>=l) {
        const width = r-l
        const area = width * Math.min(height[l], height[r])
        res = Math.max(res, area)
        if(height[l] >= height[r]) {
            r--
        } else {
            l++
        }
    }
    return res
};

File: 0011-container-with-most-water/0011-container-with-most-water.java
class Solution {
    public int maxArea(int[] height) {
        int l=0, r=height.length-1;
        int maxArea = 0;
        while(l < r) {
            int area = (r-l) * Math.min(height[l], height[r]);
            maxArea = Math.max(maxArea, area);
            if(height[r] > height[l]) {
                l++;
            } else {
                r--;
            }
        }
        return maxArea;
    }
}